src/VisualStudio/Core/Def/SampleToolWindow/SampleToolWindowFactory.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Packaging;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServices.Setup;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices
+{
+    [Export(typeof(IWpfTextViewCreationListener))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [TextViewRole(PredefinedTextViewRoles.Document)]
+    [Export(typeof(SampleToolWindowFactory))]
+    internal class SampleToolWindowFactory : IWpfTextViewCreationListener
+    {
+        [MemberNotNullWhen(true, nameof(Package))]
+        private bool Initialized { get; set; }
+
+        private RoslynPackage? Package { get; set; }
+        private IThreadingContext ThreadingContext { get; }
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public SampleToolWindowFactory(IThreadingContext threadingContext)
+        {
+            ThreadingContext = threadingContext;
+        }
+
+        public void Initialize(RoslynPackage package)
+        {
+            Package = package;
+            Initialized = true;
+        }
+
+        /// <summary>
+        /// Called whenever a new file is opened
+        /// </summary>
+        public void TextViewCreated(IWpfTextView textView)
+        {
+            ThreadingContext.JoinableTaskFactory.RunAsync(() => ShowSampleToolWindowAsync(default)).FileAndForget("Opening Sample Tool Window");
+        }
+
+        public async Task ShowSampleToolWindowAsync(CancellationToken cancellationToken)
+        {
+            if (!Initialized)
+            {
+                throw new NotSupportedException("Tool window not initialized");
+            }
```

Comment:
⚠️ In practice, I don't think we can rely on this initialization occurring via `RoslynPackage` prior to the first text view getting created. We might need to either await a force-load of the package or just return without exception during the first pass of this feature.

Reply:
Okay @sharwell I implemented returning without exception on the first pass, please let me know if that's what you had in mind. (Otherwise, I'll need help implementing a force-load of the package.)